fix(pypi): Dependency count shown in stack analysis report is incorrect

arajkumar opened the pull request:
It fixes https://github.com/fabric8-analytics/fabric8-analytics-vscode-extension/issues/315.

### Root cause:
Empty lines(with and without #) are also counted as dependencies.

### Solution:
If parsed package name is empty, ignore it.

### Cleanup:
I also did some refactoring to use JS map/reduce for a nice looking code :)

### Testing:
Chai, mocha based unit tests has been added newly. Also added [nyc](https://istanbul.js.org/) based code coverage and modified code has been covered 100%.

Follow the below steps to test this change along with fabric8-analytics-vscode-extension,

```
git clone https://github.com/fabric8-analytics/fabric8-analytics-lsp-server.git
cd fabric8-analytics-lsp-server
git fetch https://github.com/arajkumar/fabric8-analytics-lsp-server.git fix-incorrect-dep-count-pypi
git checkout FETCH_HEAD
npm install && npm run build

git clone https://github.com/fabric8-analytics/fabric8-analytics-vscode-extension.git
cd fabric8-analytics-vscode-extension
git fetch https://github.com/invincibleJai/fabric8-analytics-vscode-extension fix-py-sup-ls
git checkout FETCH_HEAD
npm install /path/to/fabric8-analytics-lsp-server/output

# Then launch VS code & debug
```

<img width="1522" alt="Screen Shot 2019-07-30 at 4 24 56 PM" src="https://user-images.githubusercontent.com/3874763/62124058-e8d4e780-b2e6-11e9-8565-78dd6fbd2fc3.png">

invincibleJai commented:
@arajkumar can even enable the test to run in cico scripts and have one commit with semantic versioning.

arajkumar commented:
@invincibleJai , Thanks for your review comments. 

>can even enable the test to run in cico scripts 

Updated the cico script to run the unit test. How to verify this on CI?

>and have one commit with semantic versioning

squashed the commit manually, but we can do that while we are merging this PR to master.

arajkumar commented:
[test]

invincibleJai commented:
[test]

arajkumar commented:
:tada: This PR is included in version 0.1.54 :tada:

The release is available on [npm package (@latest dist-tag)](https://www.npmjs.com/package/arufabric8-analytics-lsp-server)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
